numpy/_core/tests/test_stringdtype.py

```diff
@@ -1159,3 +1177,187 @@ def test_strip(string_array, unicode_array):
         np.char.strip(rjs),
         np.char.strip(rju).astype(StringDType()),
     )
+
+
+class TestImplementation:
```

Comment:
@ngoldbaum - I added an extra commit with tests of the implementation of the arena, whether it gets properly reused and released - I felt it was useful but can remove it if you'd rather not test the implementation itself.